src/js/switcher_tabs.js

```diff
@@ -0,0 +1,42 @@
+import { Dimensions } from "react-native"
+import { useAnimatedScrollHandler, useDerivedValue, interpolate } from "react-native-reanimated"
+
+export const SCREEN_WIDTH = Dimensions.get('window').width
+
+export function useSwitcherScrollHandler(translateX) {
+  return useAnimatedScrollHandler((event) => {
+    if (event.contentOffset.x < 0 || event.contentOffset.x > 4 * SCREEN_WIDTH) return
+
+    translateX.value = event.contentOffset.x
+  })
+}
+
+export function useTranslateXByScreenWidth(translateX) {
+  return useDerivedValue(() =>
+    translateX.value / SCREEN_WIDTH
+  )
+}
+
+export function useHeaderOffsetLeft(translateX, headerWidth, headerPadding) {
+  return useDerivedValue(() => {
+    let offsetLeft = 0
+
+    const headerOffsetRange = (headerWidth + (headerPadding * 2) - SCREEN_WIDTH)
+
+    if (translateX.value <= SCREEN_WIDTH * 3) {
+      offsetLeft = headerOffsetRange / 4 * (translateX.value / SCREEN_WIDTH)
+    } else {
+      offsetLeft = headerOffsetRange / 4 * (SCREEN_WIDTH * 3 / SCREEN_WIDTH)
```

Comment:
` (SCREEN_WIDTH * 3 / SCREEN_WIDTH)` ?

Reply:
If translateX exceeds the end of the 4 screens, there should be no additional translation, so instead the value should be the highest possible value, which would be `SCREEN_WIDTH * 3`

Reply:
somehow, it didn't work if i just wrote `headerOffsetRange / 12`, but it seems to work now

Reply:
Oh no, it doesn't...

Reply:
Here's with `headerOffsetRange / 12`
https://user-images.githubusercontent.com/20173411/183906006-7290658d-fd17-4882-b22b-2123e1715745.mp4

Reply:
Here's with the complicated `(SCREEN_WIDTH * 3 / SCREEN_WIDTH)`:

https://user-images.githubusercontent.com/20173411/183906106-ea8c279c-65c6-4bae-830a-305bfb110eea.mp4

Reply:
the `:bounces false` prop on the ScrollView prevents this right now, but i really don't understand why there is a difference between these two...